plugins/repository-gcs/src/main/java/org/opensearch/repositories/gcs/GoogleCloudStorageService.java

```diff
@@ -214,10 +215,12 @@ StorageOptions createStorageOptions(
             storageOptionsBuilder.setProjectId(clientSettings.getProjectId());
         }
         if (clientSettings.getCredential() == null) {
-            logger.warn(
-                "\"Application Default Credentials\" are not supported out of the box."
-                    + " Additional file system permissions have to be granted to the plugin."
-            );
+            try {
+                final GoogleCredentials credentials = SocketAccess.doPrivilegedIOException(() -> GoogleCredentials.getApplicationDefault());
+                storageOptionsBuilder.setCredentials(credentials);
+            } catch(IOException e) {
+                logger.error("Error while getting \"Application Default Credentials\"", e);
```

Comment:
**[Desired]** I think it would be better to keep the exception message format consistent with the rest of the plugin: `Failed to retrieve application default credentials `